msal/src/main/java/com/microsoft/identity/client/internal/CreateApplicationResult.java

```diff
@@ -0,0 +1,32 @@
+package com.microsoft.identity.client.internal;
+
+import com.microsoft.identity.client.IPublicClientApplication;
+import com.microsoft.identity.client.exception.MsalException;
+
+public class CreateApplicationResult  {
+
+    private MsalException mMsalException;
+    private IPublicClientApplication mIPublicClientApplication;
+    private Boolean mSuccess = false;
+
+    public CreateApplicationResult(IPublicClientApplication application, MsalException exception){
+        mMsalException = exception;
+        mIPublicClientApplication = application;
+        if(application != null){
+            mSuccess = true;
+        }
+    }
+
+    public IPublicClientApplication getPublicClientApplication(){
+        return mIPublicClientApplication;
+    }
+
+    public MsalException getException(){
+        return mMsalException;
+    }
+
+    public Boolean getSuccess(){
```

Comment:
nit : we can use primitive type `boolean` here to be safe of null pointers

Reply:
Yes